build-gluster-org/scripts/generic_package.sh

```diff
@@ -0,0 +1,200 @@
+        wget https://download.gluster.org/pub/gluster/glusterfs/${series}/${version}/glusterfs-${version}.tar.gz
+else
+        echo "found ${TGZS[0]}, using it..."
+        cp ${TGZS[0]} .
+fi
+
+echo "Creating link file.."
+ln -s glusterfs-${version}.tar.gz glusterfs_${version}.orig.tar.gz
+
+echo "Untaring.."
+tar xpf glusterfs-${version}.tar.gz
+
+# Changelogs needed for building are maintained in a separate repo.
+# the repo has to be clone and updated properly so we can copy the changelogs so far.
+
+echo "Cloning the glusterfs-debian repo"
+git clone http://github.com/gluster/glusterfs-debian.git
+
+cd glusterfs-debian/
+
+git checkout -b ${flavor}-${series}-local origin/${flavor}-glusterfs-${series}
+
+if [ "$os" == "ubuntu" ]; then
+	sed -i "1i glusterfs (${version}-${os}1~${flavor}1) ${flavor}; urgency=medium\n\n * GlusterFS ${version} GA\n\n -- GlusterFS GlusterFS deb packages <deb.packages@gluster.org>  `date +"%a, %d %b %Y %T %z"`\n" debian/changelog
+elif [ "$os" == "debian" ]; then
+	sed -i "1i glusterfs (${version}-1) ${flavor}; urgency=low\n\n * GlusterFS ${version} GA\n\n -- Gluster Packager <glusterpackager@download.gluster.org>  `date +"%a, %d %b %Y %T %z"`\n" debian/changelog
+fi
+
+git commit -a -m "Glusterfs ${version} G.A (${flavor})"
+
+echo "Copying Changelog to source"
+cp -a debian ../glusterfs-${version}/
+
+echo "Building source package.."
+cd ../glusterfs-${version}
+debuild -S -sa -k${debuild_key}
+
+echo "Uploading the packages.."
+if [ "$os" == "ubuntu" ]; then
+        cd ..
+        dput ppa:gluster/glusterfs-${series} glusterfs_${version}-${os}1~${flavor}1_source.changes
+
+    echo "Done"
+    exit
+fi
+
+# we are using the same builder machine to build so we are running the "pbuilder
+# create" everytime to create the chroot according to the os and flavor we want to build.
+echo "creating chroot for ${os} ${flavor}"
+sudo pbuilder create --distribution ${flavor} --mirror ${mirror} --debootstrapopts --keyring=/usr/share/keyrings/${os}-archive-keyring.gpg
+
+echo "Building glusterfs-${version} for ${os} ${flavor} using the chroot and .dsc we created"
+
+# have to use the .dsc file inside the ${os}${flavor} folder
+sudo pbuilder build ~/${os}-${flavor}-Glusterfs-${version}/build/glusterfs_${version}-${release}.dsc | tee build.log
+
+#move the packages to packages directory.
+mv /var/cache/pbuilder/result/glusterfs*${version}-${release}*.deb ~/${os}-${flavor}-Glusterfs-${version}/packages/
+if [ "$flavor" != "stretch" ]; then
+     mv /var/cache/pbuilder/result/libg*${version}-${release}*.deb ~/${os}-${flavor}-Glusterfs-${version}/packages/
+fi
+/usr/share/debdelta/dpkg-sig -v -k ${pbuild_key} --sign builder ~/${os}-${flavor}-Glusterfs-${version}/packages/glusterfs-*${version}-${release}*.deb
+
+cd /var/www/repos/apt/debian/
+
+rm -rf pool/* dists/* db/*
+
+cp ~/conf.distributions/${series} conf/distributions
```

Comment:
Where does thoses files come from ? Should we copy them to the builder ?

Reply:
We can copy them to builder(where we run the automation) from the machines we are currently using for building.